### PR TITLE
Gradle cleanup, shrink app size

### PR DESCRIPTION
**Pull request by TacoTheDank**
### 1st commit
- Update gradle wrapper to 7.3.2 using git bash

### 2nd commit
- Fix deprecated compileSdkVersion
- Remove unused tests
- Update Java usage to 11
- Remove unnecessary META-INF from APKs
- Reorganize dependencies
- Update Kotlin 1.5.31 -> 1.6.10
- Update AGP 7.0.3 -> 7.0.4

### 3rd commit
- Enable minification without obfuscation (shrinks APK size by about half)

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -23,33 +22,43 @@ android {
...
         googlePlay {
+            dimension "version"
             buildConfigField "String", "SAUCENAO_HIDE", "\"3\""
         }
     }
     compileOptions {
-        encoding = 'UTF-8'
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        encoding "UTF-8"
```
Comment (luk1337) [inline app/build.gradle:35]: Keep single quotes?

File: app/build.gradle [hunk 2]
```diff
@@ -23,33 +22,43 @@ android {
...
+        disable 'GoogleAppIndexingWarning'
+    }
+    packagingOptions {
+        exclude 'META-INF/*'
     }
 }
 
 dependencies {
-    implementation fileTree(dir: 'libs', include: ['*.jar'])
+// Kotlin
```
Comment (luk1337) [inline app/build.gradle:51]: Add 4 spaces indentation here

File: app/build.gradle [hunk 3]
```diff
@@ -23,33 +22,43 @@ android {
...
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
+
+// AndroidX
```
Comment (luk1337) [inline app/build.gradle:54]: Add 4 spaces indentation here

File: app/build.gradle [hunk 4]
```diff
@@ -23,33 +22,43 @@ android {
...
     implementation 'androidx.appcompat:appcompat:1.4.0'
     implementation 'androidx.cardview:cardview:1.0.0'
+    implementation 'androidx.constraintlayout:constraintlayout:2.1.2'
     implementation 'androidx.core:core-ktx:1.7.0'
     implementation 'androidx.recyclerview:recyclerview:1.2.1'
-    implementation 'androidx.constraintlayout:constraintlayout:2.1.2'
 
+// Other
```
Comment (luk1337) [inline app/build.gradle:61]: Add 4 spaces indentation here

File: gradlew [hunk 5]
```diff
@@ -1,78 +1,129 @@
-#!/usr/bin/env sh
+#!/bin/sh
```
Comment (luk1337) [inline gradlew:1]: This file shouldn't be `ISO-8859` encoded. When I ran `../gradle-7.3.1/bin/gradle wrapper` locally, it regenerated it as UTF-8.

File: .gitignore [hunk 6]
```diff
@@ -1,8 +1,10 @@
...
+local.properties
+.idea/
 .DS_Store
-/build
-/captures
-.externalNativeBuild
+build/
+captures/
+.externalNativeBuild/
+app/github/
```
Comment (luk1337) [inline .gitignore:9]: app/build is missing.

Comment (TacoTheDank) [reply .gitignore:9]: This is automatically covered by `build/`.

Comment (luk1337) [reply .gitignore:9]: wow, that's kinda confusing... Maybe just restore old gitignore files...

Comment (TacoTheDank) [reply .gitignore:9]: Alright 👍

File: gradlew (hunk 5 above)
Comment (TacoTheDank) [reply gradlew:1]: That's strange 🤔 I was doing `./gradlew wrapper --gradle-version 7.3.2 --distribution-type all`

File: app/build.gradle (hunk 1 above)
Comment (TacoTheDank) [reply app/build.gradle:35]: Ok, sure

**Pull request merged**